sentry-spring/src/main/java/io/sentry/spring/RequestPayloadExtractor.java

```diff
@@ -0,0 +1,30 @@
+package io.sentry.spring;
+
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import javax.servlet.http.HttpServletRequest;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.util.StreamUtils;
+
+final class RequestPayloadExtractor {
+
+  @Nullable
+  String extract(final @NotNull HttpServletRequest request, final @NotNull SentryOptions options) {
+    // request body can be read only once from the stream
+    // original request can be replaced with CachedBodyHttpServletRequest in SentrySpringFilter
+    if (request instanceof CachedBodyHttpServletRequest) {
+      try {
+        final byte[] body = StreamUtils.copyToByteArray(request.getInputStream());
```

Comment:
To confirm: we're gettinga brand new input stream here so not moving the offset of the request stream, right?

In ASP.NET classic, IIRC, I had to keep track of the offset within the stream (which was cached, like here) so I could read from beginning to end and then set the offset back where it was.

Reply:
`getInputStream` returns a new input stream so there is no need to track offset.